appendix/matching/main.tex

```diff
@@ -0,0 +1,503 @@
+\chapter{An introduction to matching games}
+\label{app:matching}
+
+\graphicspath{{appendix/matching/paper/img/}}
+
+Matching games form a part of game theory that were formally introduced by Gale
+and Shapley in their seminal work~\cite{Gale1962}. These games allow for the
+allocation of resources and partnerships in a mathematically fair way.
+Typically, a matching game is defined by two sets of players (called parties)
+that each have preferences over at least some of the elements of the other set.
+The objective of the game is then to find a mapping between the sets of players
+in which everyone is \emph{happy enough} with their match(es).
+
+This appendix does not contain any novel mathematics, but it does offer an
+introduction to matching games and their variants. Studying this branch of
+mathematics has contributed to a significant amount of the research conducted
+for this thesis, hence its inclusion here. That research has culminated in the
+development of a Python library for solving various matching games, \matching.
+Among other uses, the \matching\ library proves instrumental in the practical
+implementation of the novel method described in Chapter~\ref{chp:kmodes}.
+
+The \matching\ library has been developed as a research tool and adheres to the
+best practices discussed in Chapter~\ref{chp:intro}. The current version of
+Matching has also been archived on Zenodo under~\doi{10.5281/zenodo.3931026}.
+Along with the source code being modularised and fully tested (using example,
+integration and property-based unit tests) with 100\% coverage, the library is
+documented extensively. Like the \edo\ library developed for the work in
+Chapter~\ref{chp:edo}, the \matching\ documentation is hosted online at
+\href{https://matching.readthedocs.io}{\nolinkurl{matching.readthedocs.io}}.
```

Comment:
What does `\nolinkurl` do?

Reply:
`\nolinkurl` just renders the text within as if it were a URL but without a destination. I'm not sure why but `hyperref` won't pick up on some changes I've made to how URLs should be formatted so this is my workaround.

A cleaner solution would be to make my own `url` command:

```tex
\renewcommand{\myurl}[1]{%
    \href{https://#1}{\nolinkurl{#1}
}
```

... but I don't think I do this anywhere else so 🤷‍♂️

Reply:
Neat. TIL.